### PR TITLE
PS-5331 : Fix instances of 'const char* const name = "foo"' to be 'st…

### PR DESCRIPTION
**Pull request by george-lorch**
…atic const constexpr char name[]= "foo"'

- Subject says it all, modernize 'const' to 'constexpr'

### CODE REVIEW

**Comment by george-lorch** [comment]
https://ps.cd.percona.com/view/5.7/job/percona-server-5.7-param/87/

File: storage/rocksdb/rdb_datadic.h [hunk 1]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
     void (Rdb_key_def::*)(Rdb_field_packing *fpi, Field *field, uchar *buf,
                           uchar **dst, Rdb_pack_field_context *pack_ctx) const;
 
-const uint RDB_INVALID_KEY_LEN = uint(-1);
+constexpr uint RDB_INVALID_KEY_LEN = uint(-1);
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr uint RDB_INVALID_KEY_LEN = uint(-1);
```

File: storage/rocksdb/rdb_datadic.h [hunk 2]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
 
 /* How much one checksum occupies when stored in the record */
-const size_t RDB_CHECKSUM_SIZE = sizeof(uint32_t);
+constexpr size_t RDB_CHECKSUM_SIZE = sizeof(uint32_t);
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_CHECKSUM_SIZE = sizeof(uint32_t);
```

File: storage/rocksdb/rdb_datadic.h [hunk 3]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
 
 /*
   How much the checksum data occupies in record, in total.
   It is storing two checksums plus 1 tag-byte.
 */
-const size_t RDB_CHECKSUM_CHUNK_SIZE = 2 * RDB_CHECKSUM_SIZE + 1;
+constexpr size_t RDB_CHECKSUM_CHUNK_SIZE = 2 * RDB_CHECKSUM_SIZE + 1;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_CHECKSUM_CHUNK_SIZE = 2 * RDB_CHECKSUM_SIZE + 1;
```

File: storage/rocksdb/rdb_datadic.h [hunk 4]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
 
 /*
   Checksum data starts from CHECKSUM_DATA_TAG which is followed by two CRC32
   checksums.
 */
-const char RDB_CHECKSUM_DATA_TAG = 0x01;
+constexpr char RDB_CHECKSUM_DATA_TAG = 0x01;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr char RDB_CHECKSUM_DATA_TAG = 0x01;
```

File: storage/rocksdb/rdb_datadic.h [hunk 5]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
 
 /*
   Unpack data is variable length. The header is 1 tag-byte plus a two byte
   length field. The length field includes the header as well.
 */
-const char RDB_UNPACK_DATA_TAG = 0x02;
-const size_t RDB_UNPACK_DATA_LEN_SIZE = sizeof(uint16_t);
-const size_t RDB_UNPACK_HEADER_SIZE =
+constexpr char RDB_UNPACK_DATA_TAG = 0x02;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr char RDB_UNPACK_DATA_TAG = 0x02;
```

File: storage/rocksdb/rdb_datadic.h [hunk 6]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
+constexpr size_t RDB_UNPACK_DATA_LEN_SIZE = sizeof(uint16_t);
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_UNPACK_DATA_LEN_SIZE = sizeof(uint16_t);
```

File: storage/rocksdb/rdb_datadic.h [hunk 7]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
+constexpr size_t RDB_UNPACK_HEADER_SIZE =
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_UNPACK_HEADER_SIZE =
```

File: storage/rocksdb/rdb_datadic.h [hunk 8]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
 
 /*
   This header format is 1 tag-byte plus a two byte length field plus a two byte
   covered bitmap. The length field includes the header size.
 */
-const char RDB_UNPACK_COVERED_DATA_TAG = 0x03;
-const size_t RDB_UNPACK_COVERED_DATA_LEN_SIZE = sizeof(uint16_t);
-const size_t RDB_COVERED_BITMAP_SIZE = sizeof(uint16_t);
-const size_t RDB_UNPACK_COVERED_HEADER_SIZE =
+constexpr char RDB_UNPACK_COVERED_DATA_TAG = 0x03;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr char RDB_UNPACK_COVERED_DATA_TAG = 0x03;
```

File: storage/rocksdb/rdb_datadic.h [hunk 9]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
+constexpr size_t RDB_UNPACK_COVERED_DATA_LEN_SIZE = sizeof(uint16_t);
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_UNPACK_COVERED_DATA_LEN_SIZE = sizeof(uint16_t);
```

File: storage/rocksdb/rdb_datadic.h [hunk 10]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
+constexpr size_t RDB_COVERED_BITMAP_SIZE = sizeof(uint16_t);
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_COVERED_BITMAP_SIZE = sizeof(uint16_t);
```

File: storage/rocksdb/rdb_datadic.h [hunk 11]
```diff
@@ -91,40 +91,40 @@ using rdb_index_field_pack_t =
...
+constexpr size_t RDB_UNPACK_COVERED_HEADER_SIZE =
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.h]:
```suggestion
const constexpr size_t RDB_UNPACK_COVERED_HEADER_SIZE =
```

File: storage/rocksdb/rdb_utils.h [hunk 12]
```diff
@@ -147,7 +147,7 @@ namespace myrocks {
 /*
   Generic constant.
 */
-const size_t RDB_MAX_HEXDUMP_LEN = 1000;
+constexpr size_t RDB_MAX_HEXDUMP_LEN = 1000;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_utils.h]:
```suggestion
const constexpr size_t RDB_MAX_HEXDUMP_LEN = 1000;
```

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by george-lorch** [comment]
done, new jenkins https://ps.cd.percona.com/view/5.7/job/percona-server-5.7-param/93/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**